crates/bevy_pbr/src/render/mesh.rs

```diff
@@ -72,6 +81,17 @@ pub const MESH_SHADER_HANDLE: Handle<Shader> = Handle::weak_from_u128(3252377289
 pub const SKINNING_HANDLE: Handle<Shader> = Handle::weak_from_u128(13215291596265391738);
 pub const MORPH_HANDLE: Handle<Shader> = Handle::weak_from_u128(970982813587607345);
 
+/// How many textures are allowed in the view bind group layout (`@group(0)`) before
+/// broader compatibility with WebGL and WebGPU is at risk, due to the minimum guaranteed
+/// values for `MAX_TEXTURE_IMAGE_UNITS` (in WebGL) and `maxSampledTexturesPerShaderStage` (in WebGPU),
+/// currently both at 16.
+///
+/// We use 10 here because it still leaves us, in a worst case scenario, with 6 textures for the other bind groups.
+///
+/// See: <https://gpuweb.github.io/gpuweb/#limits>
+#[cfg(debug_assertions)]
+pub const MESH_PIPELINE_VIEW_LAYOUT_SAFE_MAX_TEXTURES: usize = 10;
```

Comment:
Hmm I wonder what is the ideal fallback behavior for in this type of situation. Should we disable the prepasses? Have them have no effect? Any code that assumes they will work and reads from them will fail, so there might not be a relatively straightforward way to gracefully degrade the end result...

Maybe we could make this warning a crash by default, and have a flag to enable the "extended behaviors" that require a non-baseline spec?

Reply:
I think this is a larger topic for discussion. I've wondered whether/how we will handle resource limitations long-term when we actually get to exhausting them, or having to pick and choose features to get the most out of the available resources. Previously I was thinking that at least initially, naga/wgpu will complain loudly when limits are exceeded. But now it has become a reality, and I realise the possibly obvious point that users (as in those _using_ bevy applications/games) may want to pick and choose features at runtime, then we will probably have to have some management/allocation of resources and have features requesting those allocations so that they can fail gracefully rather than just panic somewhere. I think that is too big a topic and needs to be a separate discussion from this PR.

This PR makes the situation less problematic for those not using all the features. :)